### DISCUSSION TITLE
Feedback sobre o ConvertAerotriToPAR

### DISCUSSION THREAD

**Discussion by gloriadeitos** [CEPAG-UFPR/Discussions#7] [ConvertAerotriToPAR]
<p align="center">
  <img src="https://campusmap.ufpr.br/dev/cepag-modelos/img/parcerias/fls.png" alt="FLS Logo" width="80">
  <img src="https://campusmap.ufpr.br/dev/cepag-modelos/img/cepag/cepag.png" alt="CEPAG Logo" width="100">
</p>

Esta é a seção dedicada a receber o feedback sobre o **ConvertAerotriToPAR**. Se você tiver sugestões, críticas ou qualquer comentário, sinta-se à vontade para compartilhá-los aqui!

Sua opinião é muito importante para nós e ajudará a melhorar o projeto. Lembre-se de que, para participar da discussão, você precisará de uma conta do GitHub.

**Agradecemos sua colaboração e estamos ansiosos para ouvir suas ideias!**
**Atenciosamente, Equipe do CEPAG**

<table  align="center">
  <tr>
    <td align="center" valign="top" width="80"><br />
      <a href="https://youtu.be/b0e8140kZRc">
        <img src="https://campusmap.ufpr.br/dev/cepag-modelos/img/icones/youtube.png" />
      </a>
    </td>
    <td valign="top">
      <h3>Como dar feedback no GitHub</h3>
      <p>Um vídeo explicando como deixar comentários no GitHub.</p>
      <a href="https://youtu.be/ymNI24EyOL4">
        <img src="https://img.shields.io/badge/Ver%20Video-FF0000?style=for-the-badge" target="_blank" alt="Ver Video">
      </a>
    </td>
  </tr>
</table>

**Comment by lefho1906** [comment]
Versão 2.0.0.2. Durante a importação do arquivo de projeto do programa Inpho, versões com as tags 5.0.0 e 9.2.0, é necessário ignorar faixas de voo que não possui imagens no projeto. O arquivo de projeto pode comportar todas as faixas de voo, entretanto, faixas sem imagens não possuem informações válidas ocasionando um erro na importação do arquivo conforme imagem abaixo:

![Imagem do WhatsApp de 2024-11-27 à(s) 09 39 46_b6cdc24f](https://github.com/user-attachments/assets/52780ee0-f093-4144-8c5b-71e20948bff5)

Dessa forma, é indicado ajustar o código para que durante o cálculo de orientação ignore faixas que não possuem imagens válidas. Essas faixas são representadas com os primeiros 5 dígitos conforme abaixo:

![Imagem do WhatsApp de 2024-11-28 à(s) 15 19 14_d4f3215d](https://github.com/user-attachments/assets/99ed10cb-5d41-4cf3-9efc-b6a3931d5175)

Att,

Fernando - FLS.

**Reply by lefho1906** [reply]
Para resolver o bug informado pelo Fernando durante a importação do arquivo de projeto do programa Inpho, versões com as tags 5.0.0 e 9.2.0, foi implementada uma mensagem de alerta de faixas inválidas (Warning) imediatamente após detectar faixas de voo que não possuem imagens no projeto (vide imagem abaixo). 

![Screenshot_1](https://github.com/user-attachments/assets/51498e74-0f23-4b20-8a36-586c658cef18)

O processo de importação então é interrompido até que o usuário selecione um novo arquivo de projeto. Se o novo arquivo não possuir faixas invalidas, a importação ocorre sem mensagem de alerta (vide imagem abaixo).

![Screenshot_1](https://github.com/user-attachments/assets/c708c4bc-9fd5-4380-aba6-a06ce5a7aa02)

Essa nova funcionalidade está disponível na versão 2.0.0.3.

Att,

Leonardo.

**Comment by lefho1906** [comment]
Nesta data, o Fernando da FLS reportou que a versão 2.0.0.3 apresentou problemas no gerador de imagens sintéticas. O problema foi diagnosticado pelo @chrispdobb conforme abaixo:

Descrição do probblema: Ocorre em alguns casos do processamento que, na etapa da interpolação no método bilinear, a array de índices possui valores maiores que o limite da imagem. O resultado é uma mensagem de processamento finalizado, mas que, de fato, não gerou nenhuma imagem.

![image](https://github.com/user-attachments/assets/92ac1333-5d8e-4a4c-b9d4-5c62a5ff98a3)

![Imagem do WhatsApp de 2025-07-04 à(s) 10 46 21_edb237ff](https://github.com/user-attachments/assets/59ec70d1-0b18-44f7-b8d9-dbbda1cb3dbf)

O problema foi tratado e uma nova atualização do instalador da versão 2.0.0.3 foi gerada nesta data.

**Comment by lefho1906** [comment]
Nesta data, o Fernando da FLS reportou que a versão 2.0.0.3 apresentou problemas no gerador de imagens sintéticas. O problema foi diagnosticado por @lefho1906 conforme abaixo:

Quando não existe informações de distorção de lentes no arquivo PRJ (pois o projeto já foi gerado com imagens sintéticas), as informações de distorção que estão como "NA" não são atualizadas e são repassadas na instrução de script de geração de imagens (print abaixo):

<img width="880" height="245" alt="image" src="https://github.com/user-attachments/assets/2bb8c227-de5a-462a-9b1e-14328fe6baca" />

Essa informação causa um erro na execução da rotina de geração de imagens pois o programa tenta converter a informação "NA" para número (print abaixo):

<img width="830" height="344" alt="image" src="https://github.com/user-attachments/assets/662d59d5-23d2-4c5a-9c0d-8259348f0cf6" />

**Reply by lefho1906** [reply]
Em **29/08/2025** foram realizados ajustes na rotina para que seja possível substituir os valores classificados como NA para 0. Isso é necessário para evitar problemas na execução da rotina de geração de imagens sintéticas que necessita ler valores numéricos ao invés de texto (figura abaixo):

<img width="1199" height="541" alt="image" src="https://github.com/user-attachments/assets/50aea129-8025-4187-8302-f8b4dd421bbc" />

Selecionando a opção "**Force unknown values (NA) to be changed to zero (0)**", todos os valores NA são convertidos para zero (figura abaixo):

<img width="1206" height="549" alt="image" src="https://github.com/user-attachments/assets/cecdb324-cb67-431f-b81a-cb5dfc542d53" />

Em continuidade com as melhorias, foi implementada uma ferramenta que permite zerar a rotação de todas as faixas. Isso se aplica quando é necessário apenas gerar as imagens sintéticas sem rotação similar ao procedimento feito no Metashape (figura abaixo):

<img width="924" height="459" alt="image" src="https://github.com/user-attachments/assets/fee9d0e5-e3b3-47c4-aded-781e76fbd3d0" />

<img width="926" height="454" alt="image" src="https://github.com/user-attachments/assets/5db0002e-7b6b-4e1e-8680-2829ded4d88d" />

Foram também realizados alguns ajustes no código para importar diferentes arquivos da versão $PROJECT 5.0.0. Aparentemente funcional para os arquivos testados (figura abaixo): 

<img width="340" height="92" alt="image" src="https://github.com/user-attachments/assets/9b42ff8e-cdfb-4e92-a701-72e80fe5688b" />

O instalador da versão 2.0.0.3 será atualizado com essas alterações.

Leonardo @lefho1906

**Comment by lefho1906** [comment]
**Atualização do programa gerador de imagem sintética**

Em **10/02/2026** foi feita uma atualização do programa SynthImageGen (1.0.0.7) para corrigir problemas na leitura do PLANARCONFIG. Não foi realizada alteração de versionamento da rotina ConvertAerotriToPAR. Por isso, deve-se utilizar o instalador mais recente da versão 2.0.0.3.

Leonardo.
